internal/provider/data_source_aws_transit_gateway_attachment.go

```diff
@@ -101,23 +102,56 @@ func dataSourceAwsTransitGatewayAttachmentRead(ctx context.Context, d *schema.Re
 		return diag.FromErr(err)
 	}
 
-	if waitForActive && tgwAtt.State != networkmodels.HashicorpCloudNetwork20200907TGWAttachmentStateACTIVE {
-		tgwAtt, err = clients.WaitForTGWAttachmentToBeActive(ctx, client, tgwAttID, hvnID, loc, d.Timeout(schema.TimeoutDefault))
-		if err != nil {
-			return diag.FromErr(err)
-		}
-	}
-
+	// Set ID for the resource in state.
 	link := newLink(tgwAtt.Location, TgwAttachmentResourceType, tgwAtt.ID)
 	url, err := linkURL(link)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 	d.SetId(url)
 
+	// Update TF state for the resource before waiting.
 	if err := setTransitGatewayAttachmentResourceData(d, tgwAtt); err != nil {
 		return diag.FromErr(err)
 	}
 
-	return nil
+	// Skip waiting.
+	if !waitForActive || tgwAtt.State == models.HashicorpCloudNetwork20200907TGWAttachmentStateACTIVE {
+		return nil
+	}
+
+	// If it's in a state where it could later become ACTIVE, wait.
+	waitState := false
```

Comment:
Nit: Wait state is a bit confusing of a name. Can we reverse the logic and say terminalState and then define the terminal states?

Reply:
yeah sounds good